concurrent/src/main/java/io/airlift/concurrent/VmRuntimeTicker.java

```diff
@@ -0,0 +1,180 @@
+ * period and then measures the sleep duration.  If the duration is small also,
+ * the ticker is advanced; otherwise it is ignored.
+ *
+ * The pause durations are recorded in a TimeStat for monitoring.
+ */
+public class VmRuntimeTicker
+        extends Ticker
+{
+    private static final Logger log = Logger.get(VmRuntimeTicker.class);
+
+    private static final Duration DEFAULT_LOOP_SLEEP_MILLIS = new Duration(1, MILLISECONDS);
+    private static final Duration DEFAULT_MAX_SAFE_VM_PAUSE_NANOS = new Duration(100, MILLISECONDS);
+
+    private static final Supplier<Ticker> VM_RUNTIME_TICKER = Suppliers.memoize(() -> {
+        VmRuntimeTicker ticker = new VmRuntimeTicker(Ticker.systemTicker());
+        ticker.start();
+
+        // return a wrapper so shared ticker can not be stopped
+        return new Ticker()
+        {
+            @Override
+            public long read()
+            {
+                return ticker.read();
+            }
+        };
+    });
+
+    public static Ticker getVmRuntimeTicker()
+    {
+        return VM_RUNTIME_TICKER.get();
+    }
+
+    private final Ticker systemTicker;
+    private final long loopSleepMillis;
+    private final long maxSafeVmPauseNanos;
+
+    private final Thread vmRuntimeMonitorThread;
+
+    private final AtomicLong currentValue = new AtomicLong();
+
+    private final AtomicLong totalPauseNanos = new AtomicLong();
+    private final TimeStat pauseTime = new TimeStat();
+
+    // public and volatile to assure the allocation is not optimized away
+    @SuppressWarnings({"WeakerAccess", "PublicField"})
+    public volatile Long sleepObjectAllocation;
+
+    private final AtomicBoolean badSystemNanoTime = new AtomicBoolean();
+
+    public VmRuntimeTicker(Ticker systemTicker)
+    {
+        this(systemTicker, DEFAULT_LOOP_SLEEP_MILLIS, DEFAULT_MAX_SAFE_VM_PAUSE_NANOS);
+    }
+
+    public VmRuntimeTicker(Ticker systemTicker, Duration loopSleep, Duration maxSafeVmPause)
+    {
+        this.systemTicker = requireNonNull(systemTicker, "systemTicker is null");
+
+        this.loopSleepMillis = requireNonNull(loopSleep, "loopSleep is null").toMillis();
+        checkArgument(loopSleepMillis > 0, "loopSleep must be at least 1 millis second");
+
+        this.maxSafeVmPauseNanos = requireNonNull(maxSafeVmPause, "maxSafeVmPause is null").roundTo(NANOSECONDS);
+        checkArgument(maxSafeVmPauseNanos > MILLISECONDS.toNanos(loopSleepMillis) * 2, "maxSafeVmPause must be at least 2 times larger than loopSleep");
+
+        vmRuntimeMonitorThread = new Thread(this::updateLoop, "VM Runtime Monitor");
+        vmRuntimeMonitorThread.setDaemon(true);
+    }
+
+    public void start()
+    {
+        vmRuntimeMonitorThread.start();
+    }
+
+    public void stop()
+    {
+        vmRuntimeMonitorThread.interrupt();
+    }
+
+    @Managed
+    public long getTotalPauseMillis()
+    {
+        return NANOSECONDS.toMillis(totalPauseNanos.get());
```

Comment:
I don't think you need the totalPauseNanos counter. If you want to export the millis out of convenience, than just get it from the AllTime counter from pauseTime.